Sources/InkCLI/main.swift

```diff
@@ -7,22 +7,51 @@
 import Foundation
 import Ink
 
-guard CommandLine.arguments.count > 1 else {
-    print("""
-    Ink: Markdown -> HTML converter
-    -------------------------------
-    Pass a Markdown string to convert as input,
-    and HTML will be returned as output. To use
-    STDIN as input, call ink with "-" as a single
-    argument, like this: '$ ink -'.
-    """)
+if CommandLine.arguments.contains(where: { $0 == "-h" || $0 == "--help" }) {
+    print(helpMessage)
     exit(0)
 }
 
-var markdown = CommandLine.arguments[1]
+let markdown: String
+let arguments = CommandLine.arguments
 
-if markdown == "-" {
+switch arguments.count {
+case 1:
+    // no arguments, parse stdin
     markdown = AnyIterator { readLine() }.joined(separator: "\n")
+case let count where arguments[1] == "-m" || arguments[1] == "--markdown":
+    // first argument -m or --markdown, parse Markdown string
+    guard count == 3 else {
+        printError("-m, --markdown flag takes a single following argument")
+        printError(usageMessage)
+        exit(1)
+    }
+    markdown = CommandLine.arguments[2]
+case 2:
+    // single argument, parse contents of file
+    let fileUrl: URL
+    if CommandLine.arguments[1].hasPrefix("/") {
+        fileUrl = URL(fileURLWithPath: CommandLine.arguments[1], isDirectory: false)
+    } else {
+        let dir = FileManager.default.currentDirectoryPath
+        let dirUrl = URL(fileURLWithPath: dir, isDirectory: true)
+        fileUrl = dirUrl.appendingPathComponent(CommandLine.arguments[1])
+    }
+
+    do {
+        // this is 5x faster than 'let string = try String(contentsOf: fileUrl)'
+        let data = try Data(contentsOf: fileUrl)
+        markdown = String(decoding: data, as: UTF8.self)
+    } catch {
+        printError(error.localizedDescription)
+        printError(usageMessage)
+        exit(2)
```

Comment:
What do you think about making an enum with exit codes?

Reply:
After reading [this](http://www.tldp.org/LDP/abs/html/exitcodes.html), I think we should either use `exit(1)` for all errors with no enum, or start error numbering from 3 with an enum. What do you think about the case names?

Reply:
I think `exit(1)` is just fine. Or is it planned to use these exit codes somewhere?

Reply:
I think `exit(1)` would work fine, as it's not really used anywhere, and it reduces code. I originally had numbered them sequentially before really reading anything about standards for error codes.